src/grids.jl

```diff
@@ -364,13 +379,29 @@ eachgridpoint(grid::G) where {G<:AbstractGrid} = Base.OneTo(get_npoints(G,get_nr
 gaussian_weights(nlat_half::Integer) = FastGaussQuadrature.gausslegendre(2nlat_half)[2][1:nlat_half]
 
 function clenshaw_curtis_weights(nlat_half::Integer)
-    nlat = 2nlat_half
+    nlat = 2nlat_half - 1
     θs = get_colat(FullClenshawGrid,nlat_half)
     return [4sin(θj)/(nlat+1)*sum([sin(p*θj)/p for p in 1:2:nlat]) for θj in θs[1:nlat_half]]
 end
 
 get_quadrature_weights(::Type{<:FullClenshawGrid},nlat_half::Integer) = clenshaw_curtis_weights(nlat_half)
 get_quadrature_weights(::Type{<:FullGaussianGrid},nlat_half::Integer) = gaussian_weights(nlat_half)
 get_quadrature_weights(::Type{<:OctahedralGaussianGrid},nlat_half::Integer) = gaussian_weights(nlat_half)
-get_quadrature_weights(::Type{<:HEALPixGrid},nside::Integer) =
-                nside_assert(nside) ? 2/12nside^2*[min(4iring,4nside) for iring in 1:2nside+1] : nothing
\ No newline at end of file
+# get_quadrature_weights(::Type{<:HEALPixGrid},nside::Integer) =
+#                 nside_assert(nside) ? 2/12nside^2*[min(4iring,4nside) for iring in 1:2nside] : nothing
+function get_quadrature_weights(::Type{<:HEALPixGrid},nside::Integer)
+    colat = get_colat(HEALPixGrid,nside)
+    sin_colat_Δθ = zero(colat)
+    sin_colat_Δθ[1] = sin(colat[1])*colat[2]/2
+    for j in 2:length(colat)-1
+        sin_colat_Δθ[j] = sin(colat[j])*(colat[j+1]-colat[j-1])/2
+    end
+    sin_colat_Δθ[end] = sin(colat[end])*(π-colat[end])/2
```

Comment:
These are the best quadrature weights for the HEALPix grid I was sofar able to come up with.